sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/WebExceptionUtility.java

```diff
@@ -46,10 +47,10 @@ private static boolean isWebExceptionRetriableInternal(Exception ex) {
 
         // any network failure for which we are certain the request hasn't reached the service endpoint.
         if (webEx instanceof ConnectException ||
-                webEx instanceof UnknownHostException ||
-                webEx instanceof SSLHandshakeException ||
-                webEx instanceof NoRouteToHostException ||
-                webEx instanceof SSLPeerUnverifiedException) {
+            webEx instanceof UnknownHostException ||
+            webEx instanceof SSLHandshakeException ||
+            webEx instanceof NoRouteToHostException ||
+            webEx instanceof SSLPeerUnverifiedException) {
```

Comment:
WebExceptionRetryPolicy uses this method, shouldn't  we add PrematureCloseException here

Reply:
No - these exceptions all indicate that the request never made it to the service - so it is safe to retry even for writes (see the comment above the if statement). PrmeatureCloseException means a TCP reset was received before receiving the response - but it is unknown whether the request made it to the service endpoint.